third_party/proton/csrc/lib/Profiler/CuptiProfiler.cpp

```diff
@@ -284,12 +247,28 @@ void CuptiProfiler::CuptiProfilerPimpl::doStart() {
 }
 
 void CuptiProfiler::CuptiProfilerPimpl::doFlush() {
-  CUcontext cu_context = nullptr;
-  cuda::ctxGetCurrent<false>(&cu_context);
-  if (cu_context) {
+  // cuptiActivityFlushAll returns the activity records associated with all
+  // contexts/streams.
+  // This is a blocking call but it doesn’t issue any CUDA synchronization calls
+  // implicitly thus it’s not guaranteed that all activities are completed on
+  // the underlying devices.
+  // We do an "oppurtunistic" synchronization here to try to ensure that all
+  // activities are completed on the current context.
+  // If the current context is not set, we don't do any synchronization.
+  CUcontext cuContext = nullptr;
+  cuda::ctxGetCurrent<false>(&cuContext);
+  if (cuContext)
     cuda::ctxSynchronize<true>();
-  }
-  cupti::activityFlushAll<true>(CUPTI_ACTIVITY_FLAG_FLUSH_FORCED);
+  profiler.correlation.flush(
+      /*maxRetries=*/100, /*sleepMs=*/10,
```

Comment:
There isn't a documentation for it. In general increase it may get more valid records in corner cases. Unfortunately we don't support passing configurations through the command line or python function in proton. We would like to keep a very small number of profiling knobs to avoid increasing the learning curve.

I think may could use environment variables to control it though. 

I can leave it to you to do a bit investigation and add the support. What do you think?